lib/EbookSource.php

```diff
@@ -1,10 +1,61 @@
 <?
+
+use Safe\DateTimeImmutable;
+
 class EbookSource{
+	public ?int $EbookSourceId = null;
+	public ?int $EbookId = null;
 	public EbookSourceType $Type;
 	public string $Url;
 
-	public function __construct(EbookSourceType $type, string $url){
-		$this->Type = $type;
-		$this->Url = $url;
+	public static function FromTypeAndUrl(EbookSourceType $type, string $url): EbookSource{
```

Comment:
I think this function is unnecessary - if we're creating a new instance, we can just inline the contents of this function instead of having this wrapper because there is no special logic here.

Reply:
Done.